test/integration/targets/ec2_instance/tasks/termination_protection.yml

```diff
@@ -24,62 +11,60 @@
-          "tag:Name": "{{ resource_prefix }}-test-protected-instance-in-vpc"
-          "instance-state-name": "running"
-      register: presented_instance_fact
+    - name: Check the returned value for the earlier task
+      assert:
+        that:
+          - "{{ create_instance_check_mode_results.changed }}"
+          - "{{ create_instance_check_mode_results.spec.DisableApiTermination }}"
 
-    - name: "fact checkmode ec2 instance"
+    - name: Get info on instances created in the earlier task
       ec2_instance_info:
         filters:
-          "tag:Name": "{{ resource_prefix }}-test-protected-instance-in-vpc-checkmode"
-      register: checkmode_instance_fact
+          "tag:Name": "{{ resource_prefix }}-instance-termination-protection-check-mode"
+      register: ec2_instance_info_results
 
-    - name: "Confirm whether the check mode is working normally."
+    - name: Confirm check mode does not actually create the instance
       assert:
         that:
-          - "{{ presented_instance_fact.instances | length }} > 0"
-          - "'{{ presented_instance_fact.instances.0.state.name }}' in ['running', 'pending']"
-          - "{{ checkmode_instance_fact.instances | length }} == 0"
+          - "{{ ec2_instance_info_results.instances | length }} == 0"
 
-    - name: Try to terminate the instance
+    - name: Create instance with termination protection
       ec2_instance:
-        state: absent
-        name: "{{ resource_prefix }}-test-protected-instance-in-vpc"
+        name: "{{ resource_prefix }}-instance-termination-protection"
         image_id: "{{ ec2_ami_image }}"
         tags:
           TestId: "{{ resource_prefix }}"
         security_groups: "{{ sg.group_id }}"
         vpc_subnet_id: "{{ testing_subnet_b.subnet.id }}"
         termination_protection: true
         instance_type: "{{ ec2_instance_type }}"
-      register: bad_terminate
-      ignore_errors: yes
-    - name: Cannot terminate protected instance
+        state: running
+        wait: yes
+      register: create_instance_results
+
+    - name: Check return values of the create instance task
       assert:
         that:
-          - bad_terminate is failed
-    - name: Alter termination protection setting
+          - "{{ create_instance_results.instances | length }} > 0"
+          - "'{{ create_instance_results.instances.0.state.name }}' in ['running', 'pending']"
+          - "'{{ create_instance_results.spec.DisableApiTermination }}'"
+
+    - name: Set termination protection to false for the instance
       ec2_instance:
-        name: "{{ resource_prefix }}-test-protected-instance-in-vpc"
+        name: "{{ resource_prefix }}-instance-termination-protection"
         image_id: "{{ ec2_ami_image }}"
         vpc_subnet_id: "{{ testing_subnet_b.subnet.id }}"
         termination_protection: false
         instance_type: "{{ ec2_instance_type }}"
-    - name: Try to terminate the instance again (should work)
+      register: alter_instance_results
```

Comment:
Just for completeness we should probably run _info and check that it reflects that termination protection has been unset.

Reply:
I wanted to do that. But ec2_instance_info does not return termination protection status. Here is the output from an ec2_instance_info task - https://pastebin.com/QdKJwnTG

Reply:
That's a shame